tests/Fixer/FunctionNotation/NativeFunctionInvocationFixerTest.php

```diff
@@ -0,0 +1,409 @@
+     */
+    public function testFixWithDefaultConfiguration($expected, $input = null)
+    {
+        $this->doTest($expected, $input);
+    }
+
+    /**
+     * @return array
+     */
+    public function provideCasesWithDefaultConfiguration()
+    {
+        return array(
+            array(
+'<?php
+
+json_encode($foo);
+',
+            ),
+            array(
+'<?php
+
+class WithoutNamespace
+{
+    public function bar($foo)
+    {
+        return json_encode($foo);
+    }
+}
+',
+            ),
+            array(
+'<?php
+
+namespace OneNamespaceWithBraces {}
+
+json_encode($foo);
+
+namespace AnotherNamespaceWithBraces {}
+',
+            ),
+            array(
+'<?php
+
+namespace WithoutClassPrefixed;
+
+if (isset($foo)) {
+    \json_encode($foo);
+}
+',
+            ),
+            array(
+'<?php
+
+namespace WithoutClassNotPrefixed;
+
+if (isset($foo)) {
+    \json_encode($foo);
+}
+',
+'<?php
+
+namespace WithoutClassNotPrefixed;
+
+if (isset($foo)) {
+    json_encode($foo);
+}
+',
+            ),
+            array(
+'<?php
+
+namespace Foo;
+
+class WithClassPrefixed
+{
+    public function baz($foo)
+    {
+        if (isset($foo)) {
+            \json_encode($foo);
+        }
+    }
+}',
+            ),
+            array(
+'<?php
+
+namespace WithClassNotPrefixed;
+
+class Bar
+{
+    public function baz($foo)
+    {
+        if (isset($foo)) {
+            \json_encode($foo);
+        }
+    }
+}',
+'<?php
+
+namespace WithClassNotPrefixed;
+
+class Bar
+{
+    public function baz($foo)
+    {
+        if (isset($foo)) {
+            json_encode($foo);
+        }
+    }
+}',
+            ),
+            array(
+'<?php
+
+namespace OneNamespaceWithBraces {}
+
+namespace WithoutClassInNamespaceWithBracesPrefixed
+{
+    if (isset($foo)) {
+        \json_encode($foo);
+    }
+}',
+            ),
+            array(
+'<?php
+
+namespace OneNamespaceWithBraces {}
+
+namespace WithoutClassInNamespaceWithBracesNotPrefixed
+{
+    if (isset($foo)) {
+        \json_encode($foo);
+    }
+}
+',
+'<?php
+
+namespace OneNamespaceWithBraces {}
+
+namespace WithoutClassInNamespaceWithBracesNotPrefixed
+{
+    if (isset($foo)) {
+        json_encode($foo);
+    }
+}
+',
+            ),
+        );
+    }
+
+    /**
+     * @dataProvider provideCasesWithConfiguredExclude
+     *
+     * @param string      $expected
+     * @param null|string $input
+     */
+    public function testFixWithConfiguredExclude($expected, $input = null)
+    {
+        $this->fixer->configure(array(
+            'exclude' => array(
+                'json_encode',
+            ),
+        ));
+
+        $this->doTest($expected, $input);
+    }
+
+    /**
+     * @return array
+     */
+    public function provideCasesWithConfiguredExclude()
+    {
+        return array(
+            array(
+'<?php
+
+json_encode($foo);
+',
+            ),
+            array(
+'<?php
+
+class WithoutNamespace
+{
+    public function bar($foo)
+    {
+        return json_encode($foo);
```

Comment:
I figured that if the code isn't in a namespace, then it's of no use adding the `\`, as we are already in the root namespace. That is, an attempt to resolve a function name in the same namespace will already yield the actual function, then, as it is the root namespace.

What do you think?